cns/middlewares/k8sSwiftV2_linux.go

```diff
@@ -0,0 +1,124 @@
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/middlewares/utils"
+	"github.com/pkg/errors"
+)
+
+// setRoutes sets the routes for podIPInfo used in SWIFT V2 scenario.
+func (m *K8sSWIFTv2Middleware) setRoutes(podIPInfo *cns.PodIpInfo) error {
+	logger.Printf("[SWIFTv2Middleware] set routes for pod with nic type : %s", podIPInfo.NICType)
+	podIPInfo.Routes = []cns.Route{}
+	switch podIPInfo.NICType {
+	case cns.DelegatedVMNIC:
+		virtualGWRoute := cns.Route{
+			IPAddress: fmt.Sprintf("%s/%d", virtualGW, prefixLength),
+		}
+		// default route via SWIFT v2 interface
+		route := cns.Route{
+			IPAddress:        "0.0.0.0/0",
+			GatewayIPAddress: virtualGW,
+		}
+		podIPInfo.Routes = []cns.Route{virtualGWRoute, route}
+	case cns.InfraNIC:
+		// Get and parse infraVNETCIDRs from env
+		infraVNETCIDRs, err := configuration.InfraVNETCIDRs()
+		if err != nil {
+			return errors.Wrapf(err, "failed to get infraVNETCIDRs from env")
+		}
+		infraVNETCIDRsv4, infraVNETCIDRsv6, err := utils.ParseCIDRs(infraVNETCIDRs)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse infraVNETCIDRs")
+		}
+
+		// Get and parse podCIDRs from env
+		podCIDRs, err := configuration.PodCIDRs()
+		if err != nil {
+			return errors.Wrapf(err, "failed to get podCIDRs from env")
+		}
+		podCIDRsV4, podCIDRv6, err := utils.ParseCIDRs(podCIDRs)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse podCIDRs")
+		}
+
+		// Get and parse serviceCIDRs from env
+		serviceCIDRs, err := configuration.ServiceCIDRs()
+		if err != nil {
+			return errors.Wrapf(err, "failed to get serviceCIDRs from env")
+		}
+		serviceCIDRsV4, serviceCIDRsV6, err := utils.ParseCIDRs(serviceCIDRs)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse serviceCIDRs")
+		}
+		// Check if the podIPInfo is IPv4 or IPv6
+		ip, err := netip.ParseAddr(podIPInfo.PodIPConfig.IPAddress)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse podIPConfig IP address %s", podIPInfo.PodIPConfig.IPAddress)
+		}
+		if ip.Is4() {
+			// routes for IPv4 podCIDR traffic
+			for _, podCIDRv4 := range podCIDRsV4 {
+				podCIDRv4Route := cns.Route{
+					IPAddress:        podCIDRv4,
+					GatewayIPAddress: overlayGatewayv4,
+				}
+				podIPInfo.Routes = append(podIPInfo.Routes, podCIDRv4Route)
+			}
+			// route for IPv4 serviceCIDR traffic
+			for _, serviceCIDRv4 := range serviceCIDRsV4 {
+				serviceCIDRv4Route := cns.Route{
+					IPAddress:        serviceCIDRv4,
+					GatewayIPAddress: overlayGatewayv4,
+				}
+				podIPInfo.Routes = append(podIPInfo.Routes, serviceCIDRv4Route)
+			}
+			// route for IPv4 infraVNETCIDR traffic
+			for _, infraVNETCIDRv4 := range infraVNETCIDRsv4 {
+				infraVNETCIDRv4Route := cns.Route{
+					IPAddress:        infraVNETCIDRv4,
+					GatewayIPAddress: overlayGatewayv4,
+				}
+				podIPInfo.Routes = append(podIPInfo.Routes, infraVNETCIDRv4Route)
+			}
+		} else {
```

Comment:
This "else" clauses is hanging out here and hard to spot. The logic on both sides of the conditional is the same, just with different data (`podCIDRv4` vs `podCIDRv6`). I'd recommend extracting a function.